### PR TITLE
level cap/sync should be taken into account when charming mobs

### PR DESCRIPTION
**Pull request by WiggoCaptures**
When doing things like a lvl cap 20 bcnm we shouldn't be comparing charm lvl as if we are 75 to calculate charm chance.

### CODE REVIEW

**Comment by teschnei** [comment]
Should probably check that main job is actually bst

**Comment by Wiggo32** [comment]
But... we'd want All jobs to be restricted from using BST Highest Level Obtained... not just BST... no?

File: src/map/utils/battleutils.cpp [hunk 1]
```diff
@@ -4098,6 +4098,10 @@ namespace battleutils
             {
                 charmerBSTlevel = charmerBRDlevel;
             }
+            if (charmerBSTlevel > PCharmer->GetMLevel()) // takes level cap/sync into account
```
Comment (TeoTwawki) [inline src/map/utils/battleutils.cpp]: `PCharmer->GetMLevel())` is already contained in the variable `charmerLvl`

this already gets factored into charm chance if you look at whats happening with the variable `base`.

**Comment by Wiggo32** [comment]
@teschnei We don't want to check for BST main job there like the check does for BRD up above it because every job (which has the ability to use FULL BST lvl for charm calculation regardless of BST being a subjob or not) should be restricted upon having a level cap restriction. Otherwise, people would just go into "Wings of Fury" as RDM/BST and still be able to charm the bats with a charm rate as if using a lvl 75 BST. (or whatever lvl your BST has currently achieved)

**Comment by teschnei** [comment]
okay, charm works more strangely than I thought - can you remove the comment, though? this fix also fixes people levelling as /bst (same problem), but that case has nothing to do with level sync/cap, so the comment is kind of misleading

**Comment by Wiggo32** [comment]
comment removed

**Pull request merged**